### ISSUE TITLE
Let user prefix an attribute to force insertion into `tags` of envelope.

### ISSUE THREAD

**Issue by twitchax** [frigus02/opentelemetry-application-insights#25]
So, wondering how you would feel about creating some sort of convention where an attribute key is pushed into the `tags` of the envelope.

For example, I would like to pull the `ai.user.id` generated by the web client and add it to the span for my web requests.  I know you already have some mappings, but it would be cool if a user could force it into `tags` with a prefix.

Maybe something like `KeyValue::new("tag-ai.user.id")` (or whatever), and then in `get_tags_for_span`, after all of the other mappings, we just iterate through all of the attributes, grab the ones with the prefix, and then add the suffix+value into the `tags` field of the envelope.

Thoughts?

**Comment by twitchax** [comment]
I'm also cool to implement it: I just don't want to work on it unless you are cool with that sort of functionality.

**Comment by frigus02** [comment]
I think where possible, I would prefer to use [semantic conventions](https://github.com/open-telemetry/opentelemetry-specification/tree/master/specification/trace/semantic_conventions). But I don't think there is a good match for all of the different App Insights tags.

So yeah, I think a prefix like you proposed would be a good idea. In a way all tags already have a prefix: `ai.`. I wonder if that's enough. 🤔 

We could then also offer constants for these keys, similar to the [opentelemetry-semantic-conventions](https://docs.rs/opentelemetry-semantic-conventions/0.2.0/opentelemetry_semantic_conventions/) crate. E.g.:

```rust
pub const USER_ID: Key = Key::from_static_str("optionalprefix.ai.user.id");
```

Then users can use them like:

```rust
use opentelemetry_application_insights::tags as ai;

let _span = tracer
    .span_builder("span-name")
    .with_attributes(vec![
        ai::USER_ID.string("42"),
        ai::USER_AUTH_USER_ID.string("42"),
    ])
    .start(&tracer);
```

We might want to exclude some tags like `ai.operation.id`, though. Otherwise users could accidentally break the relationship of spans.

**Comment by twitchax** [comment]
Cool.  I also agree that sticking to the semantic conventions as much as possible is the best way to go.  This really just came out of the fact that I don't _quite_ see a good semantic convention for `ai.session.id`, and I would like to capture `ai.user.id`, and `enduser.id` is already used for a different AI field.

On the prefix front, `ai.` could be a good prefix, since this crate is AI-specific anyway.  I like that idea.

Same thing on the constants: I like that idea, as well.

As for `ai.operation.id`, we could just "flip" the order.  Apply the `ai.` tags first, and then apply the ones that you currently apply (thereby overwriting the user-defined ones).

**Comment by frigus02** [comment]
Sounds like a good approach. It would be amazing if you could give it go 🙂

**Comment by frigus02** [comment]
I just published this under version 0.8.0. Thanks for adding this 👍
